wgsl/index.bs

```diff
@@ -5237,6 +5237,61 @@ If any of the parameters are out of bounds, then the call to `textureStore()`
 does nothing.
 
 
+### `textureDimensions` ### {#texturedimensions}
+
+Returns the dimensions of a texture, or texture's mip level in texels.
+
+```rust
+textureDimensions(t : texture_1d<T>) -> i32
+textureDimensions(t : texture_1d_array<T>) -> i32
+textureDimensions(t : texture_2d<T>) -> vec2<i32>
+textureDimensions(t : texture_2d<T>, level : i32) -> vec2<i32>
+textureDimensions(t : texture_2d_array<T>) -> vec2<i32>
+textureDimensions(t : texture_2d_array<T>, level : i32) -> vec2<i32>
+textureDimensions(t : texture_3d<T>) -> vec3<i32>
+textureDimensions(t : texture_3d<T>, level : i32) -> vec3<i32>
+textureDimensions(t : texture_cube<T>) -> vec3<i32>
+textureDimensions(t : texture_cube<T>, level : i32) -> vec3<i32>
+textureDimensions(t : texture_cube_array<T>) -> vec3<i32>
+textureDimensions(t : texture_cube_array<T>, level : i32) -> vec3<i32>
+textureDimensions(t : texture_multisampled_2d<T>)-> vec2<i32>
+textureDimensions(t : texture_multisampled_2d_array<T>)-> vec2<i32>
+textureDimensions(t : texture_depth_2d) -> vec2<i32>
+textureDimensions(t : texture_depth_2d, level : i32) -> vec2<i32>
+textureDimensions(t : texture_depth_2d_array) -> vec2<i32>
+textureDimensions(t : texture_depth_2d_array, level : i32) -> vec2<i32>
+textureDimensions(t : texture_depth_cube) -> vec3<i32>
+textureDimensions(t : texture_depth_cube, level : i32) -> vec3<i32>
+textureDimensions(t : texture_depth_cube_array) -> vec3<i32>
+textureDimensions(t : texture_depth_cube_array, level : i32) -> vec3<i32>
+textureDimensions(t : texture_storage_ro_1d<F>) -> i32
+textureDimensions(t : texture_storage_ro_1d_array<F>) -> i32
+textureDimensions(t : texture_storage_ro_2d<F>) -> vec2<i32>
+textureDimensions(t : texture_storage_ro_2d_array<F>) -> vec2<i32>
+textureDimensions(t : texture_storage_ro_3d<F>) -> vec3<i32>
+textureDimensions(t : texture_storage_wo_1d<F>) -> i32
+textureDimensions(t : texture_storage_wo_1d_array<F>) -> i32
+textureDimensions(t : texture_storage_wo_2d<F>) -> vec2<i32>
+textureDimensions(t : texture_storage_wo_2d_array<F>) -> vec2<i32>
+textureDimensions(t : texture_storage_wo_3d<F>) -> vec3<i32>
+```
+
+**Parameters:**
+
+<table class='data'>
+  <tr><td>`t`<td>
+  The [sampled](#sampled-texture-type),
+  [multisampled](#multisampled-texture-type), [depth](#texture-depth),
+  [read-only storage](#texture-ro) or [write-only storage](#texture-wo) texture.
+  <tr><td>`level`<td>
+  The mip level, with level 0 containing a full size version of the texture.<br>
+  If omitted, the dimensions of level 0 are returned.
```

Comment:
do we have to provide 2 versions of it? Passing `0` would not be a huge ergonomic hit if required.

Reply:
You're then touching upon the discussion in https://github.com/gpuweb/gpuweb/pull/1301 - i.e. are you suggesting we have level arguments for texture types that _currently_ do not support mipmaps (1D, etc)?
I really don't like the idea of creating a new API with pointless function arguments - it's really confusing for the developer, and there are no guarantees that those future-proofing arguments will ever get used.

If you're suggesting only having `level` parameters for those texture types that currently do support mipmaps, then we can't remove the no-level overloads if we later add mipmap support without breaking the API.

Neither approach seems favourable.

Again - my ideal solution to all this is to stick with a `Level` naming suffix for all texture functions that take a level. It allows us to have both explicit and implicit level forms, is future proof for adding mipmap support to existing textures, and also avoids overload collisions.

Reply:
I see, thanks for clarifying!